x-pack/plugins/security_solution/server/lib/entity_analytics/risk_score/routes/entity_calculation.ts

```diff
@@ -0,0 +1,153 @@
+        const securityContext = await context.securitySolution;
+
+        securityContext.getAuditLogger()?.log({
+          message: 'User triggered custom manual scoring',
+          event: {
+            action: RiskScoreAuditActions.RISK_ENGINE_ENTITY_MANUAL_SCORING,
+            category: AUDIT_CATEGORY.DATABASE,
+            type: AUDIT_TYPE.CHANGE,
+            outcome: AUDIT_OUTCOME.UNKNOWN,
+          },
+        });
+
+        const coreContext = await context.core;
+        const securityConfig = await securityContext.getConfig();
+        const siemResponse = buildSiemResponse(response);
+        const soClient = coreContext.savedObjects.client;
+
+        const riskScoreService = buildRiskScoreServiceForRequest(
+          securityContext,
+          coreContext,
+          logger
+        );
+
+        const { identifier_type: identifierType, identifier } = request.body;
+
+        try {
+          const entityAnalyticsConfig = await riskScoreService.getConfigurationWithDefaults(
+            securityConfig.entityAnalytics
+          );
+
+          if (entityAnalyticsConfig == null) {
+            return siemResponse.error({
+              statusCode: 405,
+              body: 'No Risk engine configuration found',
+            });
+          }
+
+          const {
+            dataViewId,
+            enabled,
+            range: configuredRange,
+            pageSize,
+            alertSampleSizePerShard,
+          } = entityAnalyticsConfig;
+
+          if (!enabled) {
+            return siemResponse.error({
+              statusCode: 405,
+              body: 'Risk engine is disabled',
+            });
+          }
+
+          const { index, runtimeMappings } = await getRiskInputsIndex({
+            dataViewId,
+            logger,
+            soClient,
+          });
+
+          const range = convertRangeToISO(configuredRange);
+
+          const afterKeys: AfterKeys = {};
+
+          const result: CalculateAndPersistScoresResponse =
+            await riskScoreService.calculateAndPersistScores({
+              pageSize,
+              identifierType,
+              index,
+              filter: { term: { [getFieldForIdentifier(identifierType)]: identifier } },
+              range,
+              runtimeMappings,
+              weights: [],
+              alertSampleSizePerShard,
+              afterKeys,
+              returnScores: true,
+            });
+
+          if (result.errors.length) {
+            return siemResponse.error({
+              statusCode: 500,
+              body: {
+                message: 'Error calculating the risk score for an entity.',
+                full_error: JSON.stringify(result.errors),
+              },
+              bypassErrorFormat: true,
+            });
+          }
+
+          const score =
+            result.scores_written === 1 ? result.scores?.[identifierType]?.[0] : undefined;
```

Comment:
If the risk score calculation succeeds but **no** risk score is written, the returned `score` will be `undefined`.
Is it desired?

Reply:
Yes!